numpy/wind_statistics/wind_statistics.py

```diff
@@ -0,0 +1,148 @@
+   starts on January 1 1961) for the first 52 weeks. This can be done without
+   any for loop.
+
+Bonus Bonus
+~~~~~~~~~~~
+
+Calculate the mean windspeed for each month without using a for loop.
+(Hint: look at `searchsorted` and `add.reduceat`.)
+
+Notes
+~~~~~
+
+These data were analyzed in detail in the following article:
+
+   Haslett, J. and Raftery, A. E. (1989). Space-time Modelling with
+   Long-memory Dependence: Assessing Ireland's Wind Power Resource
+   (with Discussion). Applied Statistics 38, 1-50.
+
+
+See :ref:`wind-statistics-solution`.
+"""
+
+from numpy import loadtxt, min, max, mean, std, argmax, where
+import numpy as np
+import os
+YEAR=0
+MONTH=1
+DAY=2
+LOC1=3
+
+root=os.path.dirname(__file__)
+file_name='wind.data'
+file_path=os.path.join(root,file_name)
+wind_statistics=loadtxt(file_path, dtype=float)
+np.set_printoptions(formatter={'float': '{: 0.2f}'.format})#I found this on Stack Exchange
+
+print("Statistics for windspeeds\n")
+
+print("All locations & days")
+print("Minimum: {}".format(wind_statistics[:,LOC1:].min()))
+print("Maximum: {}".format(wind_statistics[:,LOC1:].max()))
+indexOfMax = np.unravel_index(wind_statistics[:,LOC1:].argmax(),wind_statistics.shape)[0]
+print("Date recorded: {:.0f}-{:.0f}-{:.0f}".format(wind_statistics[indexOfMax,MONTH],wind_statistics[indexOfMax,DAY],wind_statistics[indexOfMax,YEAR]))
+print("Mean: {:.2f}".format(wind_statistics[:,LOC1:].mean()))
+print("Standard deviation: {}\n".format(wind_statistics[:,LOC1:].std()))
+
+print("\nEach location & all days")
+print("Minimums: {}".format(wind_statistics[:,LOC1:].min(axis=0)))
+print("Maximums: {}".format(wind_statistics[:,LOC1:].max(axis=0)))
+print("Means: {}".format(wind_statistics[:,LOC1:].mean(axis=0)))
+print("Standard deviatons: {}\n".format(wind_statistics[:,LOC1:].std(axis=0)))
+
+
+print("\nAll locations & each day")
+print("Minumums: {}".format(wind_statistics[:,LOC1:].min(axis=1)))
+print("Maximums: {}".format(wind_statistics[:,LOC1:].max(axis=1)))
+print("Means: {}".format(wind_statistics[:,LOC1:].mean(axis=1)))
+print("Standard deviatons: {}\n".format(wind_statistics[:,LOC1:].std(axis=1)))
+
+print("\nLocation with the greatest windspeed each day: {}\n".format(wind_statistics[:,LOC1:].argmax(axis=1)))
+
+maskJanuary = wind_statistics[:,MONTH]==1
+print("\nThe means of the windspeeds for each location in January: {}\n".format(wind_statistics[maskJanuary,LOC1:].mean(axis=0)))
+
+#Bonus
+
+avg_windspeed_permonth=[]
+months = np.unique(wind_statistics[:,np.array([YEAR,MONTH])],axis=0, return_index=True)
+
+first_day=0
+for last_day in months[1][1:]:
+    avg_windspeed_permonth.append(wind_statistics[first_day:last_day,LOC1:].mean())
+    first_day=last_day
+print("\nThe average windspeeeds for each month: {}\n".format(np.array(avg_windspeed_permonth)))
+
+
+print("\nThe statistics for the first 52 weeks")
+maskYear1 = wind_statistics[:,YEAR]==wind_statistics[0,0]
```

Comment:
The problem is  asking for is the _weekly_ average.  What you're reporting here is the _daily_ average.  You can do this fairly easily by reshaping the data portion of the array to be seven times its original number of columns and computing along the rows.  I _think_ it works out that we have an even number of weeks in the data set.  If not, you can just trim the data to a length that is a multiple of 7.